rm-datasource/src/main/java/io/seata/rm/datasource/exec/UpdateExecutor.java

```diff
@@ -66,20 +66,16 @@ protected TableRecords beforeImage() throws SQLException {
 
     private String buildBeforeImageSQL(TableMeta tableMeta, ArrayList<List<Object>> paramAppenderList) {
         SQLUpdateRecognizer recognizer = (SQLUpdateRecognizer)sqlRecognizer;
-        List<String> updateColumns = recognizer.getUpdateColumns();
         StringBuilder prefix = new StringBuilder("SELECT ");
-        if (!containsPK(updateColumns)) {
-            prefix.append(getColumnNameInSQL(tableMeta.getEscapePkName(getDbType()))).append(", ");
-        }
         StringBuilder suffix = new StringBuilder(" FROM ").append(getFromTableInSQL());
         String whereCondition = buildWhereCondition(recognizer, paramAppenderList);
         if (StringUtils.isNotBlank(whereCondition)) {
             suffix.append(" WHERE ").append(whereCondition);
         }
         suffix.append(" FOR UPDATE");
         StringJoiner selectSQLJoin = new StringJoiner(", ", prefix.toString(), suffix.toString());
-        for (String updateColumn : updateColumns) {
-            selectSQLJoin.add(updateColumn);
+        for (String column : tableMeta.getAllColumns().keySet()) {
+            selectSQLJoin.add(column);
```

Comment:
Columns from tableMeta, need to deal with keyword

Reply:
great found, fixed.

Reply:
solve update times cloumn